apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.TargetedSessionPoolTests.cs

```diff
@@ -419,29 +418,26 @@ public async Task WaitForPoolAsync_CancelOneOfTwo()
                 pool.MaintainPool();
                 var client = (SessionTestingSpannerClient)pool.Client;
 
-                await client.Scheduler.RunAsync(async () =>
-                {
-                    var timeBefore = client.Clock.GetCurrentDateTimeUtc();
-                    var cts = new CancellationTokenSource();
-                    var task1 = pool.WaitForPoolAsync(cts.Token);
-                    var task2 = pool.WaitForPoolAsync(default);
-
-                    await client.Scheduler.Delay(TimeSpan.FromSeconds(8));
-                    
-                    Assert.Equal(TaskStatus.WaitingForActivation, task1.Status);
-                    Assert.Equal(TaskStatus.WaitingForActivation, task2.Status);
-
-                    // If we cancel one cancelation token, that task will fail,
-                    // but the other won't.
-                    cts.Cancel();
-                    await client.Scheduler.Delay(TimeSpan.FromSeconds(1));
-                    Thread.Sleep(1000);
-                    Assert.Equal(TaskStatus.Canceled, task1.Status);
-                    Assert.Equal(TaskStatus.WaitingForActivation, task2.Status);
-
-                    // The uncancelled task completes normally
-                    await task2;
-                });
+                var timeBefore = client.Clock.GetCurrentDateTimeUtc();
+                var cts = new CancellationTokenSource();
+                var task1 = pool.WaitForPoolAsync(cts.Token);
+                var task2 = pool.WaitForPoolAsync(default);
+
+                await client.Scheduler.RunAndPauseForSeconds(8);
+
+                Assert.Equal(TaskStatus.WaitingForActivation, task1.Status);
+                Assert.Equal(TaskStatus.WaitingForActivation, task2.Status);
+                
+                // If we cancel one cancelation token, that task will fail,
+                // but the other won't.
+                cts.Cancel();
+                await client.Scheduler.RunAndPauseForSeconds(1);
+                Assert.Equal(TaskStatus.Canceled, task1.Status);
+                Assert.Equal(TaskStatus.WaitingForActivation, task2.Status);
+
+                // One second later (i.e. 10 seconds after starting), the uncancelled task completes normally
+                await client.Scheduler.RunAndPauseForSeconds(1);
+                Assert.Equal(TaskStatus.RanToCompletion, task2.Status);
```

Comment:
And this is the only test that is failing in AppVeyor because of:
Assert.Equal() Failure
Expected: RanToCompletion
Actual:   WaitingForActivation
I think is the same thing as the other. We are expecting them to be done at 10 seconds exactly, and I think we should wait till 16 seconds.